server-backend/src/utils.ts

```diff
@@ -1,5 +1,5 @@
 export function sleep(millis: number, abortSignal?: AbortSignal): Promise<void> {
-    const timeoutPromise = new Promise<void>(r => setTimeout(r, millis));
+    const timeoutPromise = new Promise<void>(r => setTimeout(r, process.env.TEST_MODE ? 1000 : millis));
```

Comment:
```suggestion
    const safeTestingTime = 1000; // This allow us to correctly execute integration tests
    const timeoutPromise = new Promise<void>(r => setTimeout(r, process.env.IS_TESTING ? safeTestingTime : millis));
```

@amanzag Maybe something like this? or do you still prefer to tackle this edge case in the function implementation?

Reply:
```suggestion
    const safeTestingTime = Math.min(1000, millis); // This allow us to correctly execute integration tests
    const timeoutPromise = new Promise<void>(r => setTimeout(r, process.env.IS_TESTING ? safeTestingTime : millis));
```
how about something like this? We should allow to sleep for less than a second.

Reply:
lgtm